go/consensus/tendermint/api/context.go

```diff
@@ -251,6 +253,16 @@ func (c *Context) IsSimulation() bool {
 	return c.mode == ContextSimulateTx
 }
 
+// SetIsMessageExecution sets the message execution flag.
+func (c *Context) SetIsMessageExecution(b bool) {
```

Comment:
Maybe it would be better if we do it like `WithMessageExecution` (similar to WithCallerAddress/WithSimulation) so that it forces the caller to use a child context and not modify the flag by accident.

Reply:
Yeah agreed, will do.